### PR TITLE
Waf rules

### PR DESCRIPTION
**Pull request by oscarDovao**
This PR contains the fastly waf rules implementation as a data source. Test coverage has been implemented but without the use of active rules function, thus minimising test dependencies 

I've not included tdhe "v1" to the file name nor the datasource terraform's reference to be consistent with the existing datasource for IP ranges, however the rest of resources' references contain "v1". 

I don't have a strong opinion about this, so open for discussion...

### CODE REVIEW

File: fastly/data_source_waf_rules_test.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
+package fastly
+
+import (
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"reflect"
+	"testing"
```
Comment (phamann) [inline fastly/data_source_waf_rules_test.go]: As above, please separate standard library imports from others.

File: fastly/data_source_waf_rules.go [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+
+	for i, r := range ruleList {
+
+		var latestRevisionNumber int
+		latestRevision, err := determineLatestRuleRevision(r.Revisions)
+		if err != nil {
+			latestRevisionNumber = 1
+		} else {
+			latestRevisionNumber = latestRevision.Revision
+		}
```
Comment (phamann) [inline fastly/data_source_waf_rules.go]: I find this easier to read:
```
latestRevisionNumber := 1
if latestRevision, err := determineLatestRuleRevision(r.Revisions); err == nil {
	latestRevisionNumber = latestRevision.Revision
}
```

File: fastly/data_source_waf_rules.go [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+
+import (
+	"errors"
+	"fmt"
+	gofastly "github.com/fastly/go-fastly/fastly"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/hashcode"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"log"
+	"sort"
+	"strconv"
```
[the remark this answers is not in the document]
Comment (oscarDovao) [reply fastly/data_source_waf_rules.go]: Imports reorganised

File: fastly/data_source_waf_rules_test.go (hunk 1 above)
Comment (oscarDovao) [reply fastly/data_source_waf_rules_test.go]: Imports reorganised

File: fastly/data_source_waf_rules.go (hunk 2 above)
Comment (oscarDovao) [reply fastly/data_source_waf_rules.go]: I absolutely agree, thanks

File: fastly/data_source_waf_rules_test.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+			},
+			local:   3,
+			Errored: false,
+		},
+	}
+
+	for _, c := range cases {
+		out, err := determineLatestRuleRevision(c.remote)
+		if (err == nil) == c.Errored {
+			t.Fatalf("Error expected to be %v but wan't", c.Errored)
```
Comment (phamann) [inline fastly/data_source_waf_rules_test.go]:
```suggestion
			t.Fatalf("Error expected to be %v but wasn't", c.Errored)
```

**Pull request merged**